### PR TITLE
Remove django-rq

### PR DESCRIPTION
**Pull request by glennmatthews**
# Closes: #2523
# What's Changed

- Remove `django-rq` dependency and all code making use of it or `rq`.
- Remove sections of documentation that are no longer needed
- Update sections of documentation that still need to mention it for historical and migration reasons.

# TODO
- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- n/a Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- n/a Example Plugin Updates (when adding/changing features)
- n/a Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Comment by itdependsnetworks** [comment]
Is this the time to update the application stack diagram?

**Comment by glennmatthews** [comment]
> Is this the time to update the application stack diagram?

Definitely - good catch.

File: nautobot/docs/installation/services.md [hunk 1]
```diff
@@ -144,6 +147,9 @@ WantedBy=multi-user.target
 +/- 1.1.0
     Prior to version 1.1.0, Nautobot utilized RQ as the primary background task worker. As of Nautobot 1.1.0, RQ is now *deprecated* and has been replaced with Celery. RQ can still be used by plugins for now, but will be removed in a future release. Please [migrate your deployment to utilize Celery as documented below](#migrating-to-celery-from-rq).
```
Comment (gsnider2195) [inline nautobot/docs/installation/services.md:148]: Is this still accurate or should it be changed to remove `RQ can still be used by plugins for now, but will be removed in a future release`

File: nautobot/docs/installation/services.md [hunk 2]
```diff
@@ -254,42 +260,9 @@ index f84073fb5..52baf6096 100644
 (END)
 ```
 
-If you are using plugins that use custom background tasks but have not yet made the change described above, you must run the [RQ worker concurrently with the Celery worker](#concurrent-celery-and-rq-nautobot-workers) until the plugin can be updated.
-
 !!! warning
     Failure to account for the Celery-to-RQ migration may break your custom background tasks
```
Comment (gsnider2195) [inline nautobot/docs/installation/services.md:264]:
```suggestion
    Failure to account for the RQ to Celery migration may break your custom background tasks
```

File: nautobot/core/api/views.py [hunk 3]
```diff
@@ -422,8 +420,6 @@ class StatusView(NautobotAPIVersionMixin, APIView):
                     "nautobot-version": {"type": "string"},
                     "plugins": {"type": "object"},
                     "python-version": {"type": "string"},
-                    # 2.0 TODO: remove rq-workers-running property
-                    "rq-workers-running": {"type": "integer"},
```
Comment (gsnider2195) [inline nautobot/core/api/views.py:426]: Should this API change be documented in `docs/installation/upgrading-from-nautobot-v1.md`

File: nautobot/core/api/views.py [hunk 4]
```diff
@@ -460,8 +456,6 @@ def get(self, request):
                 "nautobot-version": settings.VERSION,
                 "plugins": plugins,
                 "python-version": platform.python_version(),
-                # 2.0 TODO: remove rq-workers-running
-                "rq-workers-running": RQWorker.count(get_rq_connection("default")),
```
Comment (gsnider2195) [inline nautobot/core/api/views.py:464]: Should this API change be documented in `docs/installation/upgrading-from-nautobot-v1.md`

**Review by gsnider2195 — APPROVED** [review]

File: nautobot/docs/installation/services.md (hunk 1 above)
Comment (glennmatthews) [reply nautobot/docs/installation/services.md:148]: Good call, I'll revise this.

**Review by jathanism — APPROVED** [review]
*\*chef's kiss\**

**Pull request merged**